drivers/gpio/gpio_sifive.c

```diff
@@ -117,25 +117,34 @@ static int gpio_sifive_config(struct device *dev,
 	}
 
 	/* Configure gpio direction */
-	if (flags & GPIO_DIR_OUT) {
+	if (flags & GPIO_OUTPUT) {
 		gpio->in_en &= ~BIT(pin);
 		gpio->out_en |= BIT(pin);
 
 		/*
-		 * Account for polarity only for GPIO_DIR_OUT.
+		 * Account for polarity only for GPIO_OUTPUT.
 		 * invert register handles only output gpios
 		 */
-		if (flags & GPIO_POL_INV) {
+		if (flags & GPIO_ACTIVE_LOW) {
 			gpio->invert |= BIT(pin);
 		} else {
 			gpio->invert &= ~BIT(pin);
 		}
```

Comment:
Converting between logical and physical value is handled entirely in `include/drivers/gpio.h`. We should remove it from the driver.